src/main/java/org/sunbird/migrate/service/impl/UserMigrationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+                                      log.info("Failed to assign 'PUBLIC' role for user ID '{}'. Response: {}. Error: {}", userId, assignRole, assignRoleErrorMessage);
+                                      allOperationsSuccessful = false; // Update success flag
+                                  }
+                              } else {
+                                  log.info("Patch failed for user ID '{}'. Response: {}", userId, userPatchResponse);
+                                  allOperationsSuccessful = false; // Update success flag
+                              }
+                          }
+                      } else {
+                          log.info("Organization '{}' found for user ID '{}'. No migration needed.", targetOrgName, userId);
+                      }
+                  }
+                  orgFound = false;
+              }
+
+              if (allOperationsSuccessful) {
+                  response.setResponseCode(HttpStatus.OK);
+                  response.getParams().setStatus(Constants.SUCCESS);
+              } else {
+                  response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+                  response.getParams().setStatus(Constants.FAILED);
+              }
+
+          } else {
+              // Handle error response
+              if (updateResponse != null && Constants.CLIENT_ERROR.equalsIgnoreCase((String) updateResponse.get(Constants.RESPONSE_CODE))) {
+                  Map<String, Object> responseParams = (Map<String, Object>) updateResponse.get(Constants.PARAMS);
+                  if (MapUtils.isNotEmpty(responseParams)) {
+                      String errorMessage = (String) responseParams.get(Constants.ERROR_MESSAGE);
+                      response.getParams().setErrmsg(errorMessage);
+                  }
+                  response.setResponseCode(HttpStatus.BAD_REQUEST);
+              } else {
+                  response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+              }
+              response.getParams().setStatus(Constants.FAILED);
+              String errMsg = response.getParams().getErrmsg();
+              if (StringUtils.isEmpty(errMsg)) {
+                  errMsg = (String) ((Map<String, Object>) updateResponse.get(Constants.PARAMS)).get(Constants.ERROR_MESSAGE);
+                  errMsg = PropertiesCache.getInstance().readCustomError(errMsg);
+                  response.getParams().setErrmsg(errMsg);
+              }
+              log.error(errMsg, new Exception(errMsg));
+              return response;
+          }
+      } catch (Exception e) {
+          log.error("Error during user migration: {}", e.getMessage(), e);
+          response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+          response.getParams().setStatus(Constants.FAILED);
+          response.getParams().setErrmsg(e.getMessage());
+      }
+
+        return response;
+    }
+
+
+    private Map<String, Object> userSearchRequestBody() {
```

Comment:
this method should use fields... 
make sure your are adding fields in request body which mentions list of fields which you really needed to call migrate, assign role and extPatch APIs